### DISCUSSION TITLE
Run without Composer

### DISCUSSION THREAD

**Discussion by mdestafadilah** [senaranya/HL7#98] [General]
can you describe how to use this library without composer, but with autoload?

Thanks

**Comment by senaranya** [comment]
You can first use composer to get the package and set up all autoload files. It'll create a parent autoload.php in vendor/ directory.
You can now load this in your source file:
```php
namespace src; // Use whatever namespace you're in
require_once '../autoload/autoload.php'; // Adjust the path relative to this source-file
use Aranyasen\HL7\Message;
$msg = new Message("MSH|^~\\&|1|\rPV1|1|O|^AAAA1^^^BB|", null, true);
echo $msg->toString(true);
```
Once it works, you can start removing the files that are not needed anymore, including composer.json, composer.lock etc.

**Reply by mdestafadilah** [reply]
oke i will try first sir, thanks for your suggestions.

**Comment by mdestafadilah** [comment]
Hi sir, this is file of autoloader.php i have used before

```
<?php
/**
 * Users who do not have 'composer' to manage dependencies, include this
 * file to provide auto-loading of the classes in this library. 
 * filename: autoloader.php
 */
spl_autoload_register ( function ($class) {
	/*
	 * PSR-4 autoloader, based on PHP Framework Interop Group snippet (Under MIT License.)
	 * https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-4-autoloader-examples.md
	 */
	$prefix = "Escpos\\";
	$base_dir = __DIR__ . "/Escpos/";
	
	/* Only continue for classes in this namespace */
	$len = strlen ( $prefix );
	if (strncmp ( $prefix, $class, $len ) !== 0) {
		return;
	}
	
	/* Require the file if it exists */
	$relative_class = substr ( $class, $len );
	$file = $base_dir . str_replace ( '\\', '/', $relative_class ) . '.php';
	if (file_exists ( $file )) {
		require $file;
	}
} );
```

load thats file with

```
<?php
if ( ! defined('BASEPATH')) exit('No direct script access allowed');

 // filename: EscPos.php
  class Escpos
  {
  	public function __construct()
  	{
      require_once APPPATH . 'third_party/Mike42/autoloader.php';
  	}
  }
  
```

and i'm used like below

```
$this->load->library("EscPos.php");

try {
		// Enter the device file for your USB printer here
	  $connector = new Escpos\PrintConnectors\FilePrintConnector("/dev/usb/lp0");
		   
		/* Print a "Hello world" receipt" */
		$printer = new Escpos\Printer($connector);
		$printer -> text("Hello World!\n");
		$printer -> cut();

		/* Close printer */
		$printer -> close();
} catch (Exception $e) {
	echo "Couldn't print to this printer: " . $e -> getMessage() . "\n";
}
```
how to structure your HL7 library of class to load sir? 

because your trick is more big file to keep, can we do like i do before with autoloader.php, i'm doing this with [EscPos PHP](https://github.com/mike42/escpos-php/tree/master) like below

<img width="238" alt="28-06-2023 04-19-35" src="https://github.com/senaranya/HL7/assets/5575882/d5bd4d7c-a5ff-4de8-a886-285fe1fe90d0">

thanks any kind of help sir!!

**Reply by senaranya** [reply]
I'm not too good with autoloader internals, I can give it a try but it'll probably take me a long time. My library follows the standard composer conventions, so I guess if your autoloader.php is working for one composer package, it should work with mine as well.

**Reply by mdestafadilah** [reply]
i will try next day, with your package share i get .. ( i'm using PHP 5^)

> https://github.com/senaranya/HL7/issues/69#issuecomment-972461332
